src/objects/zcl_abapgit_object_odso.clas.abap

```diff
@@ -0,0 +1,381 @@
+  FINAL
+  CREATE PUBLIC .
+
+  PUBLIC SECTION.
+
+    INTERFACES zif_abapgit_object .
+
+    ALIASES mo_files
+      FOR zif_abapgit_object~mo_files .
+  PROTECTED SECTION.
+  PRIVATE SECTION.
+
+    METHODS:
+      clear_field
+        IMPORTING
+          iv_fieldname TYPE string
+        CHANGING
+          cs_metadata  TYPE any.
+ENDCLASS.
+
+
+CLASS zcl_abapgit_object_odso IMPLEMENTATION.
+
+
+  METHOD zif_abapgit_object~changed_by.
+
+    DATA: lv_dsonam  TYPE c LENGTH 30,
+          ls_return  TYPE bapiret2,
+          lr_details TYPE REF TO data.
+
+    FIELD-SYMBOLS: <ls_details> TYPE any,
+                   <lv_tstpnm>  TYPE any.
+
+    TRY.
+        CREATE DATA lr_details TYPE ('BAPI6116').
+      CATCH cx_sy_create_data_error.
+        zcx_abapgit_exception=>raise( |ODSO is not supported on this system| ).
+    ENDTRY.
+
+    ASSIGN lr_details->* TO <ls_details>.
+
+    lv_dsonam = ms_item-obj_name.
+
+    CALL FUNCTION 'BAPI_ODSO_GETDETAIL'
+      EXPORTING
+        odsobject = lv_dsonam
+      IMPORTING
+        details   = <ls_details>
+        return    = ls_return.
+
+    IF ls_return-type = 'E'.
+      zcx_abapgit_exception=>raise( |Error when geting changed by of ODSO: { ls_return-message }| ).
+    ENDIF.
+
+    ASSIGN COMPONENT 'TSTPNM' OF STRUCTURE <ls_details> TO <lv_tstpnm>.
+
+    rv_user = <lv_tstpnm>.
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~delete.
+
+    DATA: lv_odsonam    TYPE c LENGTH 30,
+          lv_objname    TYPE sobj_name,
+          lo_collection TYPE REF TO object,
+          lt_msg        TYPE rs_t_msg,
+          ls_msg        TYPE bal_s_msg,
+          ls_col_par    TYPE abap_parmbind,
+          lt_col_par    TYPE abap_parmbind_tab.
+
+    TRY.
+        CREATE OBJECT lo_collection TYPE ('CL_RSD_ODSO_COLLECTION').
+      CATCH cx_sy_create_data_error.
+        zcx_abapgit_exception=>raise( |ODSO is not supported on this system| ).
+    ENDTRY.
+
+    lv_odsonam    = ms_item-obj_name.
+    lv_objname = ms_item-obj_name.
+
+    ls_col_par-kind = cl_abap_objectdescr=>exporting.
+
+    ls_col_par-name = 'I_OBJNM'.
+    GET REFERENCE OF lv_objname INTO ls_col_par-value.
+    INSERT ls_col_par INTO TABLE lt_col_par.
+
+    ls_col_par-name = 'I_MODIFY'.
+    GET REFERENCE OF abap_true INTO ls_col_par-value.
+    INSERT ls_col_par INTO TABLE lt_col_par.
+
+    ls_col_par-name = 'I_DELETE'.
+    GET REFERENCE OF abap_true INTO ls_col_par-value.
+    INSERT ls_col_par INTO TABLE lt_col_par.
+
+    TRY.
+        CALL METHOD lo_collection->('ADD_TLOGO')
+          PARAMETER-TABLE lt_col_par.
+
+        CALL METHOD lo_collection->('DELETE').
+
+        CLEAR: lt_col_par, ls_col_par.
+        ls_col_par-kind = cl_abap_objectdescr=>importing.
+        ls_col_par-name = 'E_T_MSG'.
+        GET REFERENCE OF lt_msg INTO ls_col_par-value.
+        INSERT ls_col_par INTO TABLE lt_col_par.
+
+        CALL METHOD ('CL_RSO_APPLICATION_LOG')=>('APPL_LOG_MSG_READ')
+          PARAMETER-TABLE lt_col_par.
```

Comment:
What do think, would supply the parameters statically like in `is_active` simplify the code?

Reply:
Sure, I just didn't have enough time to do this, but I will.